pkgs/development/ocaml-modules/menhir/lib.nix

```diff
@@ -1,4 +1,6 @@
-{ lib, buildDunePackage, coqPackages, version ? "20231231" }:
+{ lib, buildDunePackage, coqPackages, version ? "20240715" }:
+# for some packages such as coqPackages.json, the default version
+# above must be kept in sync with the default version in coqPackages.MenhirLib
```

Comment:
This comment requires too much (and unnecessary) coupling between ocamlPackages and coqPackages. I’d recommend to put this constraint where it belongs, i.e., according to this comment in the `coqPackages.json` derivation.

Reply:
Makes sense. In fact, it's a nightmare to get the versions to align whereas it doesn't seem to be fundamentally required. Let's just copy what CompCert is doing.
If https://github.com/coq-community/coq-nix-toolbox/pull/250 confirm this works, we should be good.